zenodo/modules/spam/views.py

```diff
@@ -248,10 +251,13 @@ def safelist_admin():
             sa.func.max(Community.description).label('description')
         ).join(Community).group_by(User.id).filter(
             Community.created.between(from_date, to_date),
-            Community.deleted_at == None,
+            Community.deleted_at is None
         ).limit(max_users)
 
         for row in community_users:
+            if is_user_safelisted(row.user_id):
+                continue
```

Comment:
```suggestion
            Community.deleted_at is None,
            ~User.safelist.any(),
        ).limit(max_users)

        for row in community_users:
```

To avoid extra `N` queries after the big one :)

There's a JOIN-variant for this, but not sure if t changes much performance-wise.

Reply:
I thought it was okay since we're doing the iteration anyway, but this way there are fewer things to iterate through indeed.